android/app/src/main/kotlin/com/zulip/flutter/ZulipPlugin.kt

```diff
@@ -8,10 +8,27 @@ import android.util.Log
 import androidx.annotation.Keep
 import androidx.core.app.NotificationCompat
 import androidx.core.app.NotificationManagerCompat
+import androidx.core.graphics.drawable.IconCompat
 import io.flutter.embedding.engine.plugins.FlutterPlugin
 
 private const val TAG = "ZulipPlugin"
 
+fun toAndroidPerson(person: Person): androidx.core.app.Person {
+    return androidx.core.app.Person.Builder().apply {
+        person.iconData?.let { setIcon(IconCompat.createWithData(it, 0, it.size)) }
+        setKey(person.key)
+        setName(person.name)
+    }.build()
+}
+
+fun toPigeonPerson(person: androidx.core.app.Person): Person {
+    return Person(
+        null, // TODO: Serialize icon maybe using content URI, mData is not exposed.
```

Comment:
Currently, we don't return Person's IconData back when returning active notification (via `getActiveNotificationMessagingStyleByTag`) because we don't seem to have an API to get data bytes back.

Fortunately though it doesn't seem to be necessary because in my local testing, the avatar is correctly visibile even when same sender has multiple messages, which makes sense because here's what ends up happening:
```dart
MessagingStyle(
  user: Person('You')
  messages: [
    MessagingStyleMessage(content: '...',
      person: Person(key: id1, name: 'John', iconData: null)), // Old, got from active notif
    MessagingStyleMessage(content: '...',
      person: Person(key: id1, name: 'John', iconData: null)),  // Old, got from active notif
    MessagingStyleMessage(content: '...',
      person: Person(key: id1, name: 'John', iconData: [...])), // Latest message, newly added
  ],
  ...
)
```

So, Android seems to be okay with older messages _**not**_ having `iconData` for a specific Person.

Reply:
> In your example above, it's the same sender each time. What happens if some of the old senders are different from the one that's latest and has the icon data?

It works for old senders too, (also visible in the PR description video):
<img width="412" alt="Screenshot 2024-07-08 at 00 48 38" src="https://github.com/zulip/zulip-flutter/assets/36819268/83fcab07-fd1b-4748-8626-1ad66195c240">